core/src/main/scala/kafka/tools/JmxTool.scala

```diff
@@ -122,7 +124,26 @@ object JmxTool extends Logging {
       else
         List(null)
 
-    val names = queries.flatMap((name: ObjectName) => mbsc.queryNames(name, null).asScala)
+    val hasPatternQueries = queries.exists((name: ObjectName) => name.isPattern)
```

Comment:
`JmxTool` is a public tool that users use. We'd need to deprecate before removing. Technically the addition of the option should probably be a KIP